### PR TITLE
Deep history login home page

### PR DESCRIPTION
**Pull request by wduong**
Search functionalities done. Grabbing hundreds of repositories works. Need to fix users with less than three repositories. Need to fix repetitive pop-up authentication.

### CODE REVIEW

File: deepHistory/src/app/services/github-api-service.ts [hunk 1]
```diff
@@ -20,18 +20,45 @@ export class GithubApiService {
...
+                }
+            ).then(
+                res => {
+                    if (currentArray.length == 100) {
+                        console.log("returning from inside the recursive");
+                        this._getUserRepositoryList(owner, page_number + 1).then(res2 => {
+                            resolve(currentArray.concat(res2));
+                        });
+                    } else {
+                        console.log("returning from inside the out");
```
Comment (Puyush6889) [inline deepHistory/src/app/services/github-api-service.ts]: remove

**Comment by gesonchong** [comment]
Examples: 
image

image

image

File: deepHistory/src/app/services/github-api-service.ts [hunk 2]
```diff
@@ -20,18 +20,45 @@ export class GithubApiService {
...
+                        resolve(currentArray);
+                    }
+                },
+                err => {
+                    console.log("Error getting repo list");
```
Comment (Puyush6889) [inline deepHistory/src/app/services/github-api-service.ts]: console.log("Error getting repo list", err);

File: deepHistory/src/app/user/user.component.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export class UserComponent implements OnInit {
 
   displayRepos() {
     this.repos = this.user.repos;
+    console.log(this.repos);
```
Comment (Puyush6889) [inline deepHistory/src/app/user/user.component.ts]: remove

**Pull request merged**